scripts/cleanup_edomain_templates.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python2
+
+"""This script takes an provider and edomain as optional parameters, and
+searches for old templates on specified provider's export domain and deletes
+them. In case of no --provider parameter specified then this script
+traverse all the rhevm providers in cfme_data.
+"""
+
+import argparse
+import datetime
+from threading import Lock, Thread
+
+from ovirtsdk.api import API
+from utils.conf import cfme_data
+from utils.conf import credentials
+from utils.ssh import SSHClient
+from utils.log import logger
+from utils.wait import wait_for
+
+lock = Lock()
+
+
+def parse_cmd_line():
+    parser = argparse.ArgumentParser(argument_default=None)
+    parser.add_argument("--edomain", dest="edomain",
+                        help="Export domain for the remplate", default=None)
+    parser.add_argument("--provider", dest="provider",
+                        help="Rhevm provider (to look for in cfme_data)",
+                        default=None)
+    args = parser.parse_args()
+    return args
+
+
+def make_ssh_client(rhevip, sshname, sshpass):
+    connect_kwargs = {
+        'username': sshname,
+        'password': sshpass,
+        'hostname': rhevip
+    }
+    return SSHClient(**connect_kwargs)
+
+
+# get the domain edomain path on the rhevm
+def get_edomain_path(api, edomain):
+    edomain_id = api.storagedomains.get(edomain).get_id()
+    edomain_conn = api.storagedomains.get(edomain).storageconnections.list()[0]
+    return (edomain_conn.get_path() + '/' + edomain_id,
+            edomain_conn.get_address())
+
+
+def cleanup_empty_dir_on_edomain(path, edomainip, sshname, sshpass):
+    """Cleanup all the empty directories on the edomain/edomain_id/master/vms
+    else api calls will result in 400 Error with ovf not found,
+
+    Args:
+        api: API to chosen RHEVM provider.
+        edomain: Export domain of chosen RHEVM provider.
+        edomainip: edomainip to connect through ssh.
+        sshname: edomain ssh credentials.
+        sshpass: edomain ssh credentials.
+    """
+    print "RHEVM: Deleting the empty directories on edomain/vms file..."
+    ssh_client = make_ssh_client(edomainip, sshname, sshpass)
+    command = 'cd %s/master/vms && find . -type d -empty -delete' % path
+    exit_status, output = ssh_client.run_command(command)
+    if exit_status != 0:
+        print "RHEVM: Error while deleting the empty directories on path.."
+        print output
+
+
+def is_edomain_template_deleted(api, name, edomain):
+    """Checks for the templates delete status on edomain.
+
+    Args:
+        api: API for RHEVM.
+        name: template_name
+        edomain: Export domain of selected RHEVM provider.
+    """
+    try:
+        template = api.storagedomains.get(edomain).templates.get(name)
+        if not template:
+            return True
+        return False
+    except Exception as e:
+        logger.exception(e)
```

Comment:
Return value is not specified in case of exception. Does the code know that None means an exception?

Also you can shorten this code:
```python
try:
    return not api.storagedomains.get(edomain).templates.get(name)
except Exception as e:
    # ...
```